### DISCUSSION TITLE
[FEATURE] Speakdelay - delay in worldserver config

### DISCUSSION THREAD

**Discussion by IceNet** [azerothcore/azerothcore-wotlk#3932] [Ideas]
**Is your feature request related to a problem? Please describe.**
No.

**Describe the solution you'd like**
Set speakdelay config in worldserver config

**Additional context**
On line 799 in src/server/game/Chat/Channels/Channel.cpp.
speakDelay = 5;
It's hardcoded, make it able to set/disable delay in seconds in worldserver config

**Comment by FrancescoBorzi** [comment]
@IntelligentQuantum

**Comment by IntelligentQuantum** [comment]
![Screenshot_20200408_194331](https://user-images.githubusercontent.com/14080516/78801556-1b84c300-79d2-11ea-90fb-7762096de5ac.png)

@FrancescoBorzi i think this code is useless cause when i comment them nothing happened in channel chat, i mean 5sec dly is still on. :thinking:

**Comment by Kitzunu** [comment]
Did you test on an account that has security 1 = player?

If I understand the code correctly, if you are on a GM account it won't prevent spam

**Comment by Kitzunu** [comment]
BTW our config has this implemented, I suppose it is the same as the code above... But perhaps it is not properly implemented?
```
#
#    ChatFlood.MessageCount
#        Description: Chat flood protection, number of messages before player gets muted.
#        Default:     10 - (Enabled)
#                     0  - (Disabled)

ChatFlood.MessageCount = 10

#
#    ChatFlood.MessageDelay
#        Description: Time (in seconds) between messages to be counted into ChatFlood.MessageCount.
#        Default:     1

ChatFlood.MessageDelay = 1

#
#    ChatFlood.MuteTime
#        Description: Time (in seconds) characters get muted for violating ChatFlood.MessageCount.
#        Default:     10

ChatFlood.MuteTime = 10
```
